Content.Server/Chat/Managers/ChatManager.cs

```diff
@@ -152,15 +152,38 @@ public void TrySpeak(EntityUid source, string message, bool isWhisper = false, I
                 var isEmote = _sanitizer.TrySanitizeOutSmilies(message, owned, out var sanitized, out var emoteStr);
 
                 if (sanitized.Length != 0)
-                {
                     SendEntityChatType(owned, sanitized, isWhisper);
-                }
 
                 if (isEmote)
                     EntityMe(owned, emoteStr!);
             }
         }
 
+        public void TryEmote(EntityUid source, string message, IConsoleShell? shell = null, IPlayerSession? player = null)
+        {
+            var mindComponent = player?.ContentData()?.Mind;
+
+            if (mindComponent == null)
+            {
+                shell?.WriteError("You don't have a mind!");
+                return;
+            }
+
+            if (mindComponent.OwnedEntity is not {Valid: true} owned)
+            {
+                shell?.WriteError("You don't have an entity!");
+                return;
+            }
+
+            var isEmote = _sanitizer.TrySanitizeOutSmilies(message, mindComponent.OwnedEntity.Value, out var sanitized, out var emoteStr);
+
+            if (sanitized.Length != 0)
+                EntityMe(mindComponent.OwnedEntity.Value, sanitized);
+
+            if (isEmote)
+                EntityMe(mindComponent.OwnedEntity.Value, emoteStr!);
```

Comment:
Is the nullable suppression needed? I see it's above but the sanitiser should probably check?

Rest seems okay.

Reply:
Yeah it's needed